CaloFilters/src/FilterEcalNNTrigger_module.cc

```diff
@@ -0,0 +1,127 @@
+            fhicl::Table<MVATools::Config> caloBkgMVA            { Name("caloBkgMVA"),             Comment("MVA Configuration") };
+            fhicl::Atom<float>             minEtoTest            { Name("minEtoTest"),             Comment("Minimum Energy to run the MVA") };
+            fhicl::Atom<float>             minMVAScore           { Name("minMVAScore"),            Comment("MVA cut for signal") };
+            fhicl::Atom<int>               diagLevel             { Name("diagLevel"),              Comment("Diag Level"),0 };
+        };
+
+        explicit FilterEcalNNTrigger(const art::EDFilter::Table<Config>& config) :
+          EDFilter{config},
+          caloClusterToken_{consumes<CaloClusterCollection>(config().caloClusterCollection())},
+          caloBkgMVA_      (config().caloBkgMVA()),
+          minEtoTest_      (config().minEtoTest()),
+	  minMVAScore_     (config().minMVAScore()),
+          diagLevel_       (config().diagLevel())
+        {
+           produces<TriggerInfo>();
+        }
+
+        bool filter(art::Event& event) override;
+        void beginJob() override;
+
+
+     private:
+        art::ProductToken<CaloClusterCollection> caloClusterToken_;
+        MVATools          caloBkgMVA_;
+        float             minEtoTest_;
+        float             minMVAScore_;
+        int               diagLevel_;
+
+        bool filterClusters(const art::Handle<CaloClusterCollection>& caloClustersHandle, TriggerInfo& trigInfo);
+  };
+
+
+  void FilterEcalNNTrigger::beginJob()
+  {
+      caloBkgMVA_.initMVA();
+  }
+
+
+
+  bool FilterEcalNNTrigger::filter(art::Event& event)
+  {
+      art::Handle<CaloClusterCollection> caloClustersHandle;
+      event.getByToken(caloClusterToken_, caloClustersHandle);
+      
+      auto trigInfo = std::make_unique<TriggerInfo>();
+      bool retval   = filterClusters(caloClustersHandle, *trigInfo);
+      event.put(std::move(trigInfo));
+      
+      return retval;
+  }
+
+
+  //----------------------------------------------------------------------------------------------------------
+  bool FilterEcalNNTrigger::filterClusters(const art::Handle<CaloClusterCollection>& caloClustersHandle, TriggerInfo& trigInfo)
+  {
+       const Calorimeter& cal = *(GeomHandle<Calorimeter>());
+       const CaloClusterCollection& clusters(*caloClustersHandle);
+ 
+       bool select(false);
+       for (auto clusterIt=clusters.begin(); clusterIt != clusters.end();++clusterIt)
+       {
+          if (clusterIt->energyDep() < minEtoTest_) continue;
+
+          const auto& hits          = clusterIt->caloHitsPtrVector();
+	  const auto& neighborsId   = cal.crystal(hits[0]->crystalID()).neighbors();
+	  const auto& nneighborsId  = cal.crystal(hits[0]->crystalID()).nextNeighbors();
+	            
+	  double e9(hits[0]->energyDep()),e25(hits[0]->energyDep());
+          for (auto hit : hits)
```

Comment:
can you check the indentation of the whole file?

Reply:
Aargh, pesky editor. It's still this old issue with tab and space. Will fix it